earth_enterprise/src/common/performancelogger.cpp

```diff
@@ -18,19 +18,139 @@
 #include <unistd.h>
 #include <sys/syscall.h>
 #include <time.h>
-
+#include <vector>
 #include "common/performancelogger.h"
 #include "common/notify.h"
 #include "common/timeutils.h"
 
 using namespace std;
 using namespace getime;
 
-// Initialize static members of Profiler class
-PerformanceLogger * const PerformanceLogger::_instance = new PerformanceLogger();
+std::ofstream ioPrefFile;
+std::string ioFileName;
+std::ofstream timePrefFile;
+std::ofstream threadPrefFile;
+
+void checkForLeadingZero(
+        std::stringstream& io,
+        std::stringstream& thread,
+        std::stringstream& time,
+        const int& val) {
+    if (val > 9) return;
+    std::string temp("0");
+    io << temp;
+    thread << temp;
+    time << temp;
+}
+
+void printToSStream(
+        std::stringstream& io,
+        std::stringstream& thread,
+        std::stringstream& time,
+        const char& delimiter,
+        const int& val) {
+    io << val << delimiter;
+    thread << val << delimiter;
+    time << val << delimiter;
+}
+
+void openFiles() {
+    time_t t = time(0);
+    tm date = *localtime(&t);
+    std::stringstream io,time,thread;
+    io << "io.";
+    time << "time.";
+    thread << "thread.";
+    checkForLeadingZero(io,thread,time,date.tm_mon);
```

Comment:
I think this would be more readable, if you did something like:

```C++
char buffer[256];

// Format file names that look like <io_stast.12-25-2018-11:30:55.csv>:
strftime(buffer, sizeof(buffer), "io_stats.%m-%d-%Y-%H:%M:%S.csv", &date);
std::string io_stats_file_name(buffer);

strftime(buffer, sizeof(buffer), "timing.%m-%d-%Y-%H:%M:%S.csv", &date);
std::string timing_stats_file_name(buffer);

strftime(buffer, sizeof(buffer), "thread_stats.%m-%d-%Y-%H:%M:%S.csv", &date);
std::string thread_stats_file_name(buffer);
```

Reply:
Minor: File names sort better, and dates are less ambiguous, if they are formatted from the most significant to the least significant component.  Like: `year-month-day-hour:minute:second.etc`.

I would also think that you may want to see all the files that refer to a single run next to each other in directory listing, rather than all the files with I/O stats, then all the files with timing, and then all the files with thread stats.  I.e., I suggest naming the files like: `year-month-day-hour:minute:second-io_stats.csv`, etc.